.github/workflows/check-format.yml

```diff
@@ -56,9 +67,50 @@ jobs:
         run:  /usr/bin/git diff > format.patch
 
       - name: Archive diff as a patch if we failed
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         if: failure()
         with:
           name: format.patch
           path: |
             ${{ github.workspace }}/format.patch
+
+  apply_format:
+    if: ${{ always() && contains(join(needs.*.result, ','), 'failure') && (github.event_name == 'issue_comment' && startsWith(github.event.comment.body, '/autofix')) }}
+    needs: format_check
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout the GitHub created reference for the PR.
+      # The only way to do this is by using the "issue" number
+      # but that is really the PR number in this context.
+      # This is due to using an `issue_comment` event which
+      # is due to the GitHub Actions API that does not have
+      # a `pull_request_comment` event or something similar.
+      # This leaves us in a detached head state which is corrected
+      # in `apply-style/checkout.sh`
```

Comment:
Perhaps you've already considered this, but here's a possible way to avoid the extra checkout script:
https://github.com/actions/checkout/issues/331#issuecomment-1120113003

Reply:
Unfortunately, the [hub command is no longer available by default on the runners](https://github.com/actions/runner-images/issues/8362) so that wont work.